src/handlers/Status.h

```diff
@@ -0,0 +1,213 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2021 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RPC_ERRORCODES_H_INCLUDED
+#define RPC_ERRORCODES_H_INCLUDED
+
+#include <string>
+#include <boost/json.hpp>
+
+namespace RPC
+{
+
+// RPC Error codes from rippled
+// NOTE: Although the precise numeric values of these codes were never
+// intended to be stable, several API endpoints include the numeric values.
+// Some users came to rely on the values, meaning that renumbering would be
+// a breaking change for those users.
+//
+// We therefore treat the range of values as stable although they are not
+// and are subject to change.
+//
+// Please only append to this table. Do not "fill-in" gaps and do not re-use
+// or repurpose error code values.
+enum class Error {
+    // -1 represents codes not listed in this enumeration
+    rpcUNKNOWN = -1,
+
+    rpcSUCCESS = 0,
+
+    rpcBAD_SYNTAX = 1,
+    rpcJSON_RPC = 2,
+    rpcFORBIDDEN = 3,
+
+    // Misc failure
+    // unused                  4,
+    // unused                  5,
+    rpcNO_PERMISSION = 6,
+    rpcNO_EVENTS = 7,
+    // unused                  8,
+    rpcTOO_BUSY = 9,
+    rpcSLOW_DOWN = 10,
+    rpcHIGH_FEE = 11,
+    rpcNOT_ENABLED = 12,
+    rpcNOT_READY = 13,
+    rpcAMENDMENT_BLOCKED = 14,
+
+    // Networking
+    rpcNO_CLOSED = 15,
+    rpcNO_CURRENT = 16,
+    rpcNO_NETWORK = 17,
+    rpcNOT_SYNCED = 18,
+
+    // Ledger state
+    rpcACT_NOT_FOUND = 19,
+    // unused                  20,
+    rpcLGR_NOT_FOUND = 21,
+    rpcLGR_NOT_VALIDATED = 22,
+    rpcMASTER_DISABLED = 23,
+    // unused                  24,
+    // unused                  25,
+    // unused                  26,
+    // unused                  27,
+    // unused                  28,
+    rpcTXN_NOT_FOUND = 29,
+    // unused                  30,
+
+    // Malformed command
+    rpcINVALID_PARAMS = 31,
```

Comment:
This is a lot of copy paste. Luckily, the error codes are part of `xrpl_core`, so we can just include `ripple/protocol/ErrorCodes.h` and use them. Otherwise, every time rippled changes something, we have to change it here manually as well.

Reply:
Ah cool, I didn't think any of the RPC stuff was included. I can change these to use rippled rpc codes